dds/DCPS/Serializer.h

```diff
@@ -649,6 +649,30 @@ class OpenDDS_Dcps_Export Serializer {
    */
   bool write_delimiter(size_t size);
 
+  template <typename T>
+  bool peek(T& t)
+  {
+    // save
+    const bool good_bit = good_bit_;
```

Comment:
```suggestion
```

Good bit doesn't need to be saved and restored, especially in the way this hides the result of the peek deserialization.